internal/pkg/app/handlers_v1.go

```diff
@@ -204,13 +326,19 @@ func (a *APIManager) sendEventHandler(w http.ResponseWriter, r *http.Request) {
 		resp.Respond(ctx, w, doYaml(r))
 		return
 	}
-	tenantConfig, err := a.tenantStorer.GetConfig(ctx, *tid)
-	if err != nil {
-		log.Ctx(ctx).Error().Str("op", "sendEventHandler").Str("error", err.Error()).Msg("error getting tenant config")
-		resp := ErrorResponse(convertToApiError(ctx, err))
-		resp.Respond(ctx, w, doYaml(r))
-		return
+	tenantConfig := a.tenantCache.GetTenant(tid.Key())
```

Comment:
Should we mutex lock this section such that if there are tons of requests, they don't all update the cache at the same time? Alternatively, if we don't want to lock, maybe we can return the expired tenantConfig and kickoff a go routine to update cache in the background.

Reply:
ok

Reply:
i protected this with locks for now fro thread safety, i'm not sure ur other suggestions would really help with the performance issue you identified. i suggest we rerun your load test on the current version of the code and see how it goes

Reply:
For my comment, I was saying that we should lock line 337 to line 348 (although locking the cache map make sense too :)